v2/test/workload_api.go

```diff
@@ -0,0 +1,321 @@
+	if r != nil {
+		resp = r.ToProto(w.tb)
+	}
+
+	w.mu.Lock()
+	defer w.mu.Unlock()
+	w.respX509 = resp
+	for ch := range w.x509Chans {
+		select {
+		case ch <- resp:
+		default:
+			<-ch
+			ch <- resp
+		}
+	}
+}
+
+func (w *WorkloadAPI) SetJWTSVIDResponse(r *workload.JWTSVIDResponse) {
+	w.mu.Lock()
+	defer w.mu.Unlock()
+	if r != nil {
+		w.respJWT = r
+	}
+}
+
+func (w *WorkloadAPI) SetJWTBundle(trustDomain string, bundle []jose.JSONWebKey) {
+	w.mu.Lock()
+	defer w.mu.Unlock()
+
+	b := &bundleDoc{
+		RefreshHint: 9999,
+		Sequence:    1,
+	}
+	b.Keys = bundle
+
+	m, err := json.MarshalIndent(b, "", "    ")
+
+	if err != nil {
+		w.tb.Error(err)
+	}
+
+	w.respJWTBundles[trustDomain] = m
+	for ch := range w.jwtBundlesChans {
+		select {
+		case ch <- w.respJWTBundles:
+		default:
+			<-ch
+			ch <- w.respJWTBundles
+		}
+	}
+}
+
+type workloadAPIWrapper struct {
+	w *WorkloadAPI
+}
+
+func (w *workloadAPIWrapper) FetchX509SVID(req *workload.X509SVIDRequest, stream workload.SpiffeWorkloadAPI_FetchX509SVIDServer) error {
+	return w.w.fetchX509SVID(req, stream)
+}
+
+func (w *workloadAPIWrapper) FetchJWTSVID(ctx context.Context, req *workload.JWTSVIDRequest) (*workload.JWTSVIDResponse, error) {
+	return w.w.fetchJWTSVID(ctx, req)
+}
+
+func (w *workloadAPIWrapper) FetchJWTBundles(req *workload.JWTBundlesRequest, stream workload.SpiffeWorkloadAPI_FetchJWTBundlesServer) error {
+	return w.w.fetchJWTBundles(req, stream)
+}
+
+func (w *workloadAPIWrapper) ValidateJWTSVID(ctx context.Context, req *workload.ValidateJWTSVIDRequest) (*workload.ValidateJWTSVIDResponse, error) {
+	return w.w.validateJWTSVID(ctx, req)
+}
+
+type X509SVID struct {
+	CertChain     []*x509.Certificate
+	Key           crypto.Signer
+	FederatesWith []string
+}
+
+type X509SVIDResponse struct {
+	SVIDs            []X509SVID
+	Bundle           []*x509.Certificate
+	FederatedBundles map[string][]*x509.Certificate
+}
+
+func (r *X509SVIDResponse) ToProto(tb testing.TB) *workload.X509SVIDResponse {
+	bundle := derBlobFromCerts(r.Bundle)
+
+	pb := &workload.X509SVIDResponse{
+		FederatedBundles: make(map[string][]byte),
+	}
+	for _, svid := range r.SVIDs {
+		// The workload API should always respond with at one certificate and a
+		// private key but making this optional here is needed for some test
+		// flexibility.
+		var spiffeID string
+		if len(svid.CertChain) > 0 && len(svid.CertChain[0].URIs) > 0 {
+			spiffeID = svid.CertChain[0].URIs[0].String()
+		}
+		var keyDER []byte
+		if svid.Key != nil {
+			var err error
+			keyDER, err = x509.MarshalPKCS8PrivateKey(svid.Key)
+			require.NoError(tb, err)
+		}
+		pb.Svids = append(pb.Svids, &workload.X509SVID{
+			SpiffeId:      spiffeID,
+			X509Svid:      derBlobFromCerts(svid.CertChain),
+			X509SvidKey:   keyDER,
+			Bundle:        bundle,
+			FederatesWith: svid.FederatesWith,
+		})
+	}
+	for k, v := range r.FederatedBundles {
+		pb.FederatedBundles[k] = derBlobFromCerts(v)
+	}
+
+	return pb
+}
+
+type bundleDoc struct {
```

Comment:
You should use the jwtbundle.Bundle type here (which does not have refresh hints and sequence numbers).